### PR TITLE
use standard random number classes

### PR DESCRIPTION
**Pull request by ssiloti**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -127,13 +127,9 @@ AX_BOOST_CHRONO()
...
       [AC_MSG_ERROR(Boost.Chrono library not found. Try using --with-boost-chrono=lib)])
 
-AX_BOOST_RANDOM()
-AS_IF([test -z "$BOOST_RANDOM_LIB"],
-      [AC_MSG_ERROR(Boost.Random library not found. Try using --with-boost-random=lib)])
-
 CPPFLAGS="$BOOST_CPPFLAGS $CPPFLAGS"
 LDFLAGS="$BOOST_LDFLAGS $LDFLAGS"
-LIBS="$BOOST_CHRONO_LIB $BOOST_RANDOM_LIB $LIBS"
+LIBS="$BOOST_CHRONO_LIB $BOOST_SYSTEM_LIB $LIBS"
```
Comment (arvidn) [inline configure.ac]: this will conflict with another patch I have in the queue :( (to remove the dependency on boost.chrono)

**Comment by arvidn** [comment]
lgtm

File: configure.ac (hunk 1 above)
Comment (ssiloti) [inline configure.ac]: Eh, can't really do anything about that. It's not like it's a big deal, I won't begrudge you if you merge yours first and make me spend the 30s to rebase :)

Comment (arvidn) [inline configure.ac]: yeah, it was just a heads-up.

**Pull request merged**